src/lib/Publisher.coffee

```diff
@@ -64,28 +64,31 @@ class Publisher extends Channel
       else
         return cb("Channel is closed and will not re-open? #{@state} #{@confirm} #{@confirmState}") if cb
 
-    # data must be a buffer
-    if typeof data is 'string'
-      options.contentType = 'string/utf8'
-      data = new Buffer(data, 'utf8')
-
-    else if typeof data is 'object' and !(data instanceof Buffer)
-      if options.contentType?
-        debug 1, ()=> return "contentType specified but data isn't a buffer, #{JSON.stringify options}"
-        if cb?
-          cb("contentType specified but data isn't a buffer")
-          return
-
-      # default use JSON
-      data = new Buffer(JSON.stringify(data), 'utf8')
-      options.contentType = 'application/json'
-
-      # data = BSON.serialize data
-      # options.contentType = 'application/bson'
-
-    else if data is undefined
-      data = new Buffer(0)
-      options.contentType = 'application/undefined'
+    # check type of data only if options.contentType was not specified
+    if !options.contentType?
+
+      # data must be a buffer
+      if typeof data is 'string'
+        options.contentType = 'string/utf8'
+        data = new Buffer(data, 'utf8')
+
+      else if typeof data is 'object' and !(data instanceof Buffer)
+        if options.contentType?
```

Comment:
Looks like this condition is always false now. Maybe add some flag to make checking data and changing contentType explicitly?